src/scene/batching/batch-manager.js

```diff
@@ -943,7 +943,7 @@ class BatchManager {
         }
 
         batch2.meshInstance.castShadow = batch.meshInstance.castShadow;
-        batch2.meshInstance._shader = batch.meshInstance._shader;
+        batch2.meshInstance._shader = batch.meshInstance._shader.slice();
```

Comment:
wondering, does this fix a bug?

Reply:
a possible bug yeah, not an actual one I observed. I don't think two mesh instances should share a private array. If one gets deallocated, the other should not keep their copy.

Reply:
The before solution would save memory, as it would share the same array between two mesh instances. Now these are separate arrays. But the arrays are tiny, so I prefer the correct behaviour in all cases.